ui/src/BaseEditor.ts

```diff
@@ -30,7 +29,7 @@ export abstract class BaseEditor<ContextT> implements vscode.Disposable {
     public async showEditor(context: ContextT, sizeLimit?: number /* in Megabytes */): Promise<void> {
         const fileName: string = await this.getFilename(context);
         const resourceName: string = await this.getResourceName(context);
-        this.appendLineToOutput(localize('opening', 'Opening "{0}"...', fileName), {resourceName: resourceName});
+        this.appendLineToOutput(localize('opening', 'Opening "{0}"...', fileName), { resourceName: resourceName });
         if (sizeLimit !== undefined) {
             const size: number = await this.getSize(context);

```

Comment:
The whole point of using the VS Code fs API is [because](https://github.com/microsoft/vscode/blob/f207b3830717f2c354ea12ead37baff21f33a680/src/vs/vscode.d.ts#L7197) 
> It allows extensions to work with files from the local disk as well as files from remote places, like the remote extension host or ftp-servers.

 which you just can't get from "fs-extra" AFAIK. There are other benefits like "useTrash" during the "delete" method, which is pretty cool IMO. That being said, yes it won't be a simple drop-in replacement for "fs-extra" which has a lot of _extra_ stuff. I think we need more of a "big picture" strategy first rather than trying to replace individual fs-extra calls like this.  Something that lets us benefit from the VS Code API without making all of our code ugly in the process. For example, one potential strategy is to follow the pattern of `AzureUserInput`/`IAzureUserInput`/`TestUserInput` which wraps VS Code APIs and adds some of our own stuff on top